pulsar-functions/transforms/src/main/java/org/apache/pulsar/functions/transforms/TransformFunction.java

```diff
@@ -0,0 +1,76 @@
+package org.apache.pulsar.functions.transforms;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.pulsar.client.api.Schema;
+import org.apache.pulsar.client.api.schema.GenericObject;
+import org.apache.pulsar.functions.api.Context;
+import org.apache.pulsar.functions.api.Function;
+import org.apache.pulsar.functions.api.Record;
+
+@Slf4j
+public class TransformFunction implements Function<GenericObject, Void> {
+
+    private final List<TransformStep> steps = new ArrayList<>();
+    private final Gson gson = new Gson();
+
+    @Override
+    public void initialize(Context context) {
+        Object config = context.getUserConfigValue("steps")
+                .orElseThrow(() -> new IllegalArgumentException("missing required 'steps' parameter"));
+        LinkedList<Map<String, String>> stepsConfig;
+        try {
+            TypeToken<LinkedList<Map<String, String>>> typeToken = new TypeToken<LinkedList<Map<String, String>>>(){};
+            stepsConfig = gson.fromJson((gson.toJson(config)), typeToken.getType());
+        } catch (Exception e) {
+            throw new IllegalArgumentException("could not parse configuration", e);
+        }
+        for (Map<String, String> step : stepsConfig) {
+            String type = step.get("type");
+            if ("drop-fields".equals(type)) {
+                String fields = step.get("fields");
```

Comment:
what about moving this block to a static utility method in RemoveFieldFunction ?

Reply:
done